python_modules/dagster/dagster/core/asset_defs/assets.py

```diff
@@ -146,20 +160,43 @@ def asset_keys(self) -> AbstractSet[AssetKey]:
         return set(self.asset_keys_by_output_name.values())
 
     @property
-    def asset_keys_by_output_name(self) -> Mapping[str, AssetKey]:
+    def dependency_asset_keys(self) -> Iterable[AssetKey]:
+        return self.asset_keys_by_input_name.values()
+
+    @property
+    def optional_asset_keys(self) -> AbstractSet[AssetKey]:
+        """
+        The set of asset keys that the underlying computation can produce but is not required to
+        """
+        return set(self._asset_keys_by_output_name.values()) - self.asset_keys
+
+    @property
+    def node_asset_keys_by_output_name(self) -> Mapping[str, AssetKey]:
         return self._asset_keys_by_output_name
 
     @property
-    def asset_keys_by_input_name(self) -> Mapping[str, AssetKey]:
+    def node_asset_keys_by_input_name(self) -> Mapping[str, AssetKey]:
```

Comment:
The distinction between this and `asset_keys_by_input_name` isn't obvious from reading the name. A short comment would be helpful for disambiguating.